lib/Wikia/src/Domain/User/Preferences/LocalPreference.php

```diff
@@ -14,6 +14,12 @@ function __construct( $name, $value, $wikiId ) {
 		Assert::true( !empty( $name ), "invalid preference name" );
 		Assert::true( !empty( $wikiId ), "invalid wiki id" );
 
+		if ( $value === "true" ) {
+			$value = true;
+		} elseif ( $value === "false" ) {
+			$value = false;
+		}
```

Comment:
Sorry, I don't mean for everything, but perhaps in the case where it's a string that looks bool-like. Just a suggestion...